vllm/attention/ops/prefix_prefill.py

```diff
@@ -600,7 +609,7 @@ def _fwd_kernel_alibi(
                 v = v_load
             p = p.to(v.dtype)
 
-            acc += tl.dot(p, v, allow_tf32=False)
```

Comment:
Ok, not 100%, but very close to it. 

I checked the documentation of triton and also the code about this behavior. This flag (which is deprecated) only affects when we set the dtype of model to float32, at least the tests of kernel does not cover float32 (my guess to not increase CI time...). Anyway it looks like to me in practice we will be using float16 to serve the models, so most of cases it will not change. I was actually expecting the CI to tell me if I am breaking something or not with this change (especially on AMD cards). The first time I opened they did run fine, so I think it still make sense to keep it. Also if you see my [gist](https://gist.github.com/wallashss/072d24bfab83bd2f80584a45dcf215ce) you can see that I run these kernel tests extending dtype to float32 and all passed. 

One motivation I see to use this flag is to make it work on turing  devices, which this PR already covers by sending as a parameter, so to me it looks like safe. Set `allow_tf32=False` is equivalent to set `input_precision='ieee'` 


Ref: https://triton-lang.org/main/python-api/generated/triton.language.dot.html#triton.language.dot

Reply:
@wallashss it seems to me that for the operations that currently include `allow_tf32=False`, it would be safer to "hardcode" `input_precision="ieee"`, since as you say this is equivalent and also what's needed for the Turing workaround anyhow?

I see in the triton repo that there's various reports of precision issues from tf32, which I assume is the reason that this arg is set in the vLLM code in the first place.

Reply:
Ok, I set `input_precision="ieee"` for the kernel that had `allow_tf32=False` to keep full compatibility. 

Thanks @njhill @mgoin